File: A2/src/stmt/AssignStmt.java
package stmt;
import dictionary.*;
import exp.Exp;
import model.MyStmtException;
import model.PrgState;
import stack.*;

public class AssignStmt implements IStmt
{
	String id;
	Exp exp;
	
	public AssignStmt (String id, Exp exp)
	{
		this.id = id;
		this.exp = exp;
	}
	public String toString()
	{
		return id + "=" + exp.toString();
	}
	public PrgState execute(PrgState state) throws MyStmtException
	{
		MyIStack<IStmt> stk = state.getExeStack();
		MyIDictionary<String, Integer> symTbl = state.getSymTable();
		int val = exp.eval(symTbl);
		if (symTbl.isDefined(id))
		{
			symTbl.update(id, val);
		}
		else
		{
			symTbl.add(id, val);
		}
		return state;
	}
}

File: A7/Stmt/AssignStmt.cs
using A7.Dictionary;
using A7.Expr;
using A7.Model;
using A7.Stack;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace A7.Stmt
{
    class AssignStmt : IStmt
    {
        String id;
        Exp exp;
        public AssignStmt(String id, Exp exp)
        {
            this.id = id;
            this.exp = exp;
        }
        public override String ToString()
        {
            return id + "=" + exp.ToString();
        }
        public PrgState Execute(PrgState state)
        {
            MyIStack<IStmt> stk = state.GetExeStack();
            MyIDictionary<String, int> symTbl = state.GetSymTable();
            int val = exp.Eval(symTbl);
            if (symTbl.IsDefinedU(id))
            {
                symTbl.Update(id, val);
            }
            else
            {
                symTbl.Add(id, val);
            }
            return state;
        }
    }
}
